pkg/services/publicdashboards/models/models_test.go

```diff
@@ -15,36 +17,88 @@ func TestPublicDashboardTableName(t *testing.T) {
 
 func TestBuildTimeSettings(t *testing.T) {
 	var dashboardData = simplejson.NewFromAny(map[string]interface{}{"time": map[string]interface{}{"from": "2022-09-01T00:00:00.000Z", "to": "2022-09-01T12:00:00.000Z"}})
-	fromMs, toMs := internal.GetTimeRangeFromDashboard(t, dashboardData)
+	defaultFromMs, defaultToMs := internal.GetTimeRangeFromDashboard(t, dashboardData)
+
+	selectionFromMs := strconv.FormatInt(time.Now().UnixMilli(), 10)
+	selectionToMs := strconv.FormatInt(time.Now().Add(time.Hour).UnixMilli(), 10)
+
 	testCases := []struct {
 		name       string
 		dashboard  *models.Dashboard
 		pubdash    *PublicDashboard
 		timeResult TimeSettings
+		reqDTO     PublicDashboardQueryDTO
 	}{
 		{
 			name:      "should use dashboard time if pubdash time empty",
 			dashboard: &models.Dashboard{Data: dashboardData},
-			pubdash:   &PublicDashboard{},
+			pubdash:   &PublicDashboard{TimeSelectionEnabled: false},
 			timeResult: TimeSettings{
-				From: fromMs,
-				To:   toMs,
+				From: defaultFromMs,
+				To:   defaultToMs,
 			},
+			reqDTO: PublicDashboardQueryDTO{},
 		},
 		{
 			name:      "should use dashboard time even if pubdash time exists",
 			dashboard: &models.Dashboard{Data: dashboardData},
-			pubdash:   &PublicDashboard{TimeSettings: &TimeSettings{From: "now-12", To: "now"}},
+			pubdash:   &PublicDashboard{TimeSelectionEnabled: false, TimeSettings: &TimeSettings{From: "now-12", To: "now"}},
+			timeResult: TimeSettings{
+				From: defaultFromMs,
+				To:   defaultToMs,
+			},
+			reqDTO: PublicDashboardQueryDTO{},
+		},
+		{
+			name:      "should use selected values if time selection is enabled",
+			dashboard: &models.Dashboard{Data: dashboardData},
+			pubdash:   &PublicDashboard{TimeSelectionEnabled: true, TimeSettings: &TimeSettings{From: "now-12", To: "now"}},
+			reqDTO: PublicDashboardQueryDTO{
+				TimeRange: TimeSettings{
+					From: selectionFromMs,
+					To:   selectionToMs,
+				},
+			},
+			timeResult: TimeSettings{
+				From: selectionFromMs,
+				To:   selectionToMs,
+			},
+		},
+		{
+			name:      "should use default values if time selection is enabled but the values provided are invalid",
```

Comment:
I think this might be making API too smart. I think it would be better to return an error or something rather than make an assumption.

Reply:
I did it this way because we are doing something similar for `intervalMS` and `maxDataPoints` (overriding invalid values). Not sure what would be the best solution here. Do you think we should send an error back if the time selection is enabled and the `from` or `to` values are not provided? @owensmallwood @juanicabanas help us decide here

Reply:
I agree with @jalevin . I think an error is the best return

Reply:
If I had no knowledge of the api and I passed very invalid values for a time range, I'd be a little confused when it was successful. I think what Jeff's saying make sense.

Reply:
Great, done. Thanks for the feedback!